src/LayerEditor/ui/data/tools/selector.py

```diff
@@ -0,0 +1,23 @@
+
+
+class Selector:
+    """ Used for selecting one item(value) from list.
+        Also used to check if selected item is still in iterable object.
+        If not than select other item that is part of iterable"""
+    def __init__(self, value):
+        self.value = value
+        self.old_value = None
+
+    def validate(self, lst: list) -> bool:
+        """Check if selected item is still in the list. If not select first item in the list.
+            return: True if `value` was valid, False if value was changed."""
+        if self.value not in lst:
```

Comment:
Not sure how. For each item there should be one equal operation. It could potentialy be faster using sets, but for those it would still have to generate hash so it might not be big difference.

Reply:
- Rename validate to "make_valid".
- Remove old_value, or return from "make_valid".